src/sentry/static/sentry/app/views/dashboardsV2/widget/new/widgetNew.tsx

```diff
@@ -68,58 +91,73 @@ class WidgetNew extends AsyncView<Props, State> {
     return routeTitleGen(t('Dashboards - Widget Builder'), params.orgId, false);
   }
 
-  handleFieldChange<F extends keyof State>(field: F, value: State[F]) {
+  handleFieldChange = <F extends keyof State>(field: F, value: State[F]) => {
+    if (field === 'displayType') {
+      this.setState(state => ({
+        ...state,
+        [field]: value,
+        title: tct('Custom [displayType] Widget', {displayType: displayTypes[value]}),
+      }));
+      return;
+    }
+
     this.setState(state => ({...state, [field]: value}));
-  }
+  };
 
-  handleQueryChange = (widgetQuery: WidgetQuery, index: number) => {
+  handleEventQueryChange = (index: number, widgetQuery: WidgetQuery) => {
```

Comment:
Later on, this will need to be updated to the new widget builder behaviour I introduced in https://github.com/getsentry/sentry/pull/24279

Reply:
I will check the changes and update in the part 4 ...thanks for pointing it out 🙏